Policies/Kubernetes/fsgroup/rego/src_test.rego

```diff
@@ -0,0 +1,185 @@
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_allowed_all_with_range {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters(anyrun, [range_in_range])}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_no_fsgroup_allowed_all_with_range_outofrange {
+    input := { "review": input_review, "parameters": input_parameters(anyrun, [range_outofrange])}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_MustRunAs_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters(mustrun, [range_in_range])}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_MustRunAs_not_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters(mustrun, [range_outofrange])}
+    results := violation with input as input
+    count(results) == 1
+}
+test_input_no_fsgroup_MustRunAs_not_allowed {
+    input := { "review": input_review, "parameters": input_parameters(mustrun, [range_in_range])}
+    results := violation with input as input
+    count(results) == 1
+}
+test_input_securitycontext_no_fsgroup_MustRunAs_not_allowed {
+    input := { "review": input_review_with_securitycontext_no_fsgroup, "parameters": input_parameters(mustrun, [range_in_range])}
+    results := violation with input as input
+    count(results) == 1
+}
+test_input_fsgroup_MayRunAs_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters(mayrun, [range_in_range])}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_MayRunAs_not_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters(mayrun, [range_outofrange])}
+    results := violation with input as input
+    count(results) == 1
+}
+test_input_no_fsgroup_MayRunAs_allowed {
+    input := { "review": input_review, "parameters": input_parameters(mayrun, [range_in_range])}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_securitycontext_no_fsgroup_MayRunAs_allowed {
+    input := { "review": input_review_with_securitycontext_no_fsgroup, "parameters": input_parameters(mayrun, [range_in_range])}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_MayRunAs_two_ranges_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters(mayrun, [range_outofrange, range_high_range])}
+    results := violation with input as input
+    count(results) == 0
+}
+test_input_fsgroup_MayRunAs_two_ranges_not_allowed {
+    input := { "review": input_review_with_fsgroup, "parameters": input_parameters(mayrun, [range_outofrange, range_high_range_outofrange])}
+    results := violation with input as input
+    count(results) == 1
```

Comment:
Why is this a violation?
In test_input_fsgroup_MayRunAs_two_ranges_allowed, I see FSGroup value as 1000. Ranges are (1, 1000) and (1200, 2000).
In test_input_fsgroup_MayRunAs_two_ranges_not_allowed, I see FSGroup value as 1000. Ranges are (1, 1000) and (2100, 4000).

In both the above tests, FSGroup value 1000 falls under (1, 1000) bucket and hence violations should not be thrown.